### DISCUSSION TITLE
Disable the 0x ligature?

### DISCUSSION THREAD

**Discussion by yurikhan** [tonsky/FiraCode#1454] [Q&A]
The first demo image calls this “Hexadecimal Ex”. The examples are 0xFF and 1920x1080. The letter `x` is displayed similar to the character `×` U+00D7 Multiplication Sign.

I think this ligature is harmful for three reasons:

* It makes it hard to spot typographic mistakes in two+-dimensional sizes.
* The `x` in hexadecimal literals has no historical relationship to the diagonal cross; actually I believe it comes from the shortening of “hexadecimal” to “hex” then “ex”.
* It is inconsistent with other hexadecimal forms such as HTML/XML `&#xF00;` and string literal `\xF00`.

Therefore I’d like to disable it at least for my own use. Do I have to do my own builds?

**Comment by tonsky** [comment]
I’m afraid yes. I hear your arguments, but I think it makes a cool feature nevertheless, so I would like to keep it

**Reply by itshawk** [reply]
Is it possible to make it a variant? I'm personally finding it fairly unreadable in my code.

**Comment by billgan1024** [comment]
For anyone wondering how to do this:

- Download [fontforge](https://fontforge.org/en-US/)
- open fira code
- at the top, press View > Goto, then type in "x.multiply"
- select the glyph to highlight it in blue
- press Encoding > Detach & Remove Glyphs. 

If you did this step properly, you can view the result by going to Metrics > New Metrics window and typing "0x"...

Finally, generate a new version of firacode using File > Generate Fonts. 

PS: You can also use this to remove other ligatures by repeating this process.
